src/Microsoft.ML.TensorFlow/TensorflowTransform.cs

```diff
@@ -109,34 +104,48 @@ public static TensorFlowTransform Create(IHostEnvironment env, ModelLoadContext
             Contracts.CheckValue(env, nameof(env));
             env.CheckValue(ctx, nameof(ctx));
             ctx.CheckAtModel(GetVersionInfo());
+
             // *** Binary format ***
+            // int: indicator for frozen models
             // stream: tensorFlow model.
             // int: number of input columns
             // for each input column
             //   int: id of int column name
             // int: number of output columns
             // for each output column
             //   int: id of output column name
-            byte[] modelBytes = null;
-            if (!ctx.TryLoadBinaryStream("TFModel", r => modelBytes = r.ReadByteArray()))
-                throw env.ExceptDecode();
-            var numInputs = ctx.Reader.ReadInt32();
-            env.CheckDecode(numInputs > 0);
-            string[] inputs = new string[numInputs];
-            for (int j = 0; j < inputs.Length; j++)
-                inputs[j] = ctx.LoadNonEmptyString();
+            var isFrozen = ctx.Reader.ReadInt32();
+            if (isFrozen == 1)
+            {
+                byte[] modelBytes = null;
+                if (!ctx.TryLoadBinaryStream("TFModel", r => modelBytes = r.ReadByteArray()))
+                    throw env.ExceptDecode();
 
-            bool isMultiOutput = ctx.Header.ModelVerReadable >= 0x00010002;
-            var numOutputs = 1;
-            if (isMultiOutput)
-                numOutputs = ctx.Reader.ReadInt32();
+                var io = ModelInputsOutputs(env, ctx);
+                return new TensorFlowTransform(env, modelBytes, io.Item1, io.Item2, (isFrozen == 1));
+            }
+            else
+            {
+                // Load model binary
+                byte[] tfFilesBin = null;
+                var load = ctx.TryLoadBinaryStream("TFSavedModel", br => tfFilesBin = br.ReadByteArray());
 
-            env.CheckDecode(numOutputs > 0);
-            var outputs = new string[numOutputs];
-            for (int j = 0; j < outputs.Length; j++)
-                outputs[j] = ctx.LoadNonEmptyString();
+                if (File.Exists(TempSavedModelZipName))
+                    File.Delete(TempSavedModelZipName);
```

Comment:
>TempSavedModelZipName [](start = 32, length = 21)

As Zeeshan mentioned, this would be problematic in a CV setting, or if a pipeline contains multiple TF transforms.

Reply:
I have modified the logic for save&restore so we do things in memory.

---
In reply to: [217432460](https://github.com/dotnet/machinelearning/pull/853#discussion_r217432460) [](ancestors = 217432460)